amplipi/streams.py

```diff
@@ -738,8 +738,18 @@ def _activate(self, vsrc: int):
 
   def _deactivate(self):
     if self._is_running():
-      self.proc.kill()
-      self.proc.wait()
+      try:
+        self.proc.terminate()
+        self.proc.wait(timeout=4)
+      except:
+        # Likely a subprocess.TimeoutException, but we will handle all exceptions the same.
+        # Because this runs within a process_monitor.py, we need to handle the entire
+        # process group.
+        os.killpg(self.proc.pid, signal.SIGKILL)
+        # does the child still leave a zombie if we wait() the parent? pid space is not at
+        # a premium, so I'll leave that question unanswered.
+        self.proc.wait()
```

Comment:
Can this proc.wait block forever?

Reply:
I think it can... if `SIGKILL` fails to terminate a process. If my memory serves me, this can happen when blocking IO is occurring. FWIW I don't think we should engineer for that case (at least at present), because that's effectively very broken anyhow - chances are that blocking IO is communication is with our ALSA loopback.

We do this in other places in the code, using `communicate()` (`wait()` is communicate without the inputs & outputs); this is additionally how [the `subprocess` docs recommend accomplishing this](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate).